operators/cuda/rotary.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ocos.h"
+#include "rotary_impl.cuh"
+#include "ortx_common.h"
+
+namespace contrib {
+
+/**
+*  Y = Rotary(X) is equivalent to if side == LEFT:
+*
+*   N = X.shape[-1]
+*   Y = X.copy()
+*   Y[...,:N/2] = X[...,N/2:]
+*   Y[...,N/2:] = -X[...,:N/2]
+*
+*   And the opposite if side == RIGHT.
+*/
+template <typename T>
+struct Rotary {
+  template <typename TDict>
+  OrtxStatus OnModelAttach(const TDict& dict) {
+    std::string empty;
+    std::string side = dict.TryToGetAttributeWithDefault("side", empty);
+    if (side == "left") {
+      side_ = RotarySide::LEFT;
+    }
+    else if (side == "right") {
+      side_ = RotarySide::RIGHT;
+    }
+    else {
+      return {kOrtxErrorInvalidArgument, "side must be 'left' or 'right'."};
+    }
+
+    return {};
+  }
+  OrtxStatus Compute(Ort::Custom::CUDAKernelContext* ctx,
```

Comment:
How is `split` related to the formula above?

Reply:
The Rotary replaces something like:

```
x1, x2 = Split(X, split, axis=-1)
Y = Concat(-x2, x1, axis=-1)
```

When I implemented this kernel, I found it was equal splits on llama but I wondered if it could be not equal splits so I chose to leave this parameter so that I could still use the optimized and let the kernel fails when the split are not equal. Then I would know not equal splits must be implemented. I can remove if you know this case never happens.

Reply:
Please make sure the `exact` behavior is described in comments for this op. I don't see the use and explanation of `split` in the equations. In addition to math symbols, you can use ONNX sub-graph to describe them.

Reply:
I think we can assume even split and throw otherwise. With `split`, you will need to introduce a sub-graph to compute the right `split` from the shape of `X`. That will slows down ORT..

Reply:
I don't think so. The formula remains the same, whether or not the splits are equal or not.